snapcraft/models/assertions.py

```diff
@@ -92,7 +103,9 @@ class EditableConfdbSchemaAssertion(models.CraftBaseModel):
 
     def marshal_scalars_as_strings(self) -> dict[str, Any]:
         """Marshal the model where all scalars are represented as strings."""
-        return cast_dict_scalars_to_strings(self.marshal())
+        # Use exclude_none=True to align with test expectation that None fields are omitted.
+        marshalled = self.model_dump(mode="python", exclude_none=True)
+        return cast_dict_scalars_to_strings(marshalled)
```

Comment:
This is wrong on many levels.  It shouldn't be needed in the first place, we need the mode to be json instead of python, and we're purposefully using an upstream implementation in craft-application for marshalling.